File: ios/MenuElementsEmitter.m
//
//  MenuElementsEmitter.m
//  BlueWallet
//

#import "MenuElementsEmitter.h"

static MenuElementsEmitter *sharedInstance;

@implementation MenuElementsEmitter

RCT_EXPORT_MODULE();

+ (BOOL)requiresMainQueueSetup {
    return YES;
}

+ (instancetype)sharedInstance {
    if (!sharedInstance) {
        sharedInstance = [[self alloc] init];
    }
    return sharedInstance;
}

- (instancetype)init {
    self = [super init];
    if (self) {
        sharedInstance = self;
    }
    return self;
}

- (NSArray<NSString *> *)supportedEvents {
    return @[
        @"openSettings",
        @"addWalletMenuAction",
        @"importWalletMenuAction",
        @"reloadTransactionsMenuAction"
    ];
}

- (void)openSettings {
    [self sendEventWithName:@"openSettings" body:nil];
}

- (void)addWalletMenuAction {
    [self sendEventWithName:@"addWalletMenuAction" body:nil];
}

- (void)importWalletMenuAction {
    [self sendEventWithName:@"importWalletMenuAction" body:nil];
}

- (void)reloadTransactionsMenuAction {
    [self sendEventWithName:@"reloadTransactionsMenuAction" body:nil];
}

@end

File: ios/MenuElementsEmitter.swift
import Foundation
import React

@objc(MenuElementsEmitter)
class MenuElementsEmitter: RCTEventEmitter {

    static let sharedInstance: MenuElementsEmitter = MenuElementsEmitter()

    override class func requiresMainQueueSetup() -> Bool {
        return true
    }
    
    override func supportedEvents() -> [String]! {
        return ["openSettings", "addWalletMenuAction", "importWalletMenuAction", "reloadTransactionsMenuAction"]
    }
    
    @objc func openSettings() {
        sendEvent(withName: "openSettings", body: nil)
    }
    
    @objc func addWalletMenuAction() {
        sendEvent(withName: "addWalletMenuAction", body: nil)
    }
    
    @objc func importWalletMenuAction() {
        sendEvent(withName: "importWalletMenuAction", body: nil)
    }
    
    @objc func reloadTransactionsMenuAction() {
        sendEvent(withName: "reloadTransactionsMenuAction", body: nil)
    }
}
